res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -0,0 +1,974 @@
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter1", true);
+
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter3", true);
+    engine.softTakeover("[EqualizerRack1_[Channel2]_Effect1]", "parameter2", true);
+    engine.softTakeover("[EqualizerRack1_[Channel3]_Effect1]", "parameter1", true);
+
+    engine.softTakeover("[QuickEffectRack1_[Channel1]]", "super1", true);
+    engine.softTakeover("[QuickEffectRack1_[Channel2]]", "super1", true);
+
+    engine.softTakeover("[Master]", "crossfader", true);
+    engine.softTakeover("[Master]", "gain", true);
+    engine.softTakeover("[Master]", "headMix", true);
+    engine.softTakeover("[Master]", "headGain", true);
+
+    // Dirty hack to set initial values in the packet parser
+    var data = TraktorS2MK3.toBytes("01 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 00 00 00 00");
+    TraktorS2MK3.incomingData(data);
+};
+
+TraktorS2MK3.registerInputJog = function (message, group, name, offset, bitmask, callback) {
+    // Jog wheels have 4 byte input
+    message.addControl(group, name, offset, "I", bitmask);
+    message.setCallback(group, name, callback);
+};
+
+TraktorS2MK3.registerInputScaler = function (message, group, name, offset, bitmask, callback) {
+    message.addControl(group, name, offset, "H", bitmask);
+    message.setCallback(group, name, callback);
+};
+
+TraktorS2MK3.registerInputButton = function (message, group, name, offset, bitmask, callback) {
+    message.addControl(group, name, offset, "B", bitmask);
+    message.setCallback(group, name, callback);
+};
+
+TraktorS2MK3.playHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue(field.group, "cue_set", field.value);
+    } else {
+        var playing = engine.getValue(field.group, "play");
+        engine.setValue(field.group, "play", !playing);
+    }
+};
+
+TraktorS2MK3.cueHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue(field.group, "cue_gotoandstop", field.value);
+    } else {
+        engine.setValue(field.group, "cue_default", field.value);
+    }
+};
+
+TraktorS2MK3.shiftHandler = function (field) {
+    TraktorS2MK3.shiftPressed[field.group] = field.value;
+    engine.setValue("[Controls]", "touch_shift", field.value);
+    TraktorS2MK3.outputHandler(field.value, field.group, "shift");
+};
+
+TraktorS2MK3.keylockHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    var keylock = engine.getValue(field.group, "keylock");
+    engine.setValue(field.group, "keylock", !keylock);
+};
+
+TraktorS2MK3.syncHandler = function (field) {
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue(field.group, "beatsync_phase", field.value);
```

Comment:
Ok, but the shift behaviour is not really documented. In the wiki entry, it sounds like this depends on quantize, not the shift state.

Reply:
I forgot this in the Wiki page, updated it now.